api/v1alpha1/envoygateway_types.go

```diff
@@ -512,6 +512,14 @@ type ExtensionManager struct {
 	//
 	// +optional
 	FailOpen bool `json:"failOpen,omitempty"`
+
+	// MaxMessageSize defines the maximum message size in bytes that can be
+	// sent to or received from the Extension Service.
+	// Default: 4 MB
+	// +kubebuilder:validation:Minimum=1
+	//
+	// +optional
+	MaxMessageSize *int `json:"maxMessageSize,omitempty"`
```

Comment:
should we use `resource.Quantity` here similar to https://github.com/envoyproxy/gateway/blob/0ee0a8af8e0a86bfbafbca4cee6d5432afa07830/api/v1alpha1/connection_types.go#L28?